app/controllers/application_controller.rb

```diff
@@ -323,4 +324,8 @@ def handle_unverified_request
     logger.info("unverified_request session:" + session.inspect)
     reset_session
   end
+
+  def store_auto_publish_url
+    Thread.current[:auto_publish_url] = "#{request.protocol}#{request.host_with_port}"
```

Comment:
I think we want to include the protocol, if we use protocol relative then an http portal will always be stuck running http activities.  Also I think this URL gets used as a redirect so I'm not sure how redirects handle protocol relative. Many of our activities are https compatible so it would make sense to be able to publish those as https even though the portal is running http.  I haven't worked through the cases though so there might be gotchas there.  If there aren't gotchas then it would be best if we had a flag (probably hidden) on the activity that indicated if it was https compatible and we use that to determine what protocol to use when publishing the activity. But that should be another PR not this one.